### PR TITLE
RaceEthnicity_CASE-1192

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
Refactored Race and Ethnicity Card by pulling RaceForm and EthnicityForm from react-wood-duck

### CODE REVIEW

**Comment by SashaKyz** [comment]
Can one of the admins verify this patch?

File: app/javascript/Clients/clientIdPage/RaceEthnicity/Ethnicity.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/RaceEthnicity/Ethnicity.test.js]: I like these tests 😄.  To make them a little more easy to read when you run the specs, I'd switch the organization of the descriptions to something like:

```javascript
describe('#onchange' () => {
  describe('when value is yes', () =>
    it('sets the value of latinoOrigin and disableFields', () =>
    });
  });
});
```

File: app/javascript/Clients/clientIdPage/RaceEthnicity/EthnicityDetails.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const HISPANIC_LATINO_ORIGIN = Object.freeze([
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/RaceEthnicity/EthnicityDetails.js]: I like the freezes here.

File: app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  {}
+);
+
+const personId = '1';
+
+const getIsRaceIndeterminateValueSelector = races => {
+  const isUnknown = races.Unknown;
+  const isAbandoned = races.Abandoned;
+  const isDeclinedToAnswer = races['Declined to answer'];
+  return Boolean(isUnknown || isAbandoned || isDeclinedToAnswer);
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js]: This needs an independent test.

File: app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    this.setState({
+      races: races,
+      raceDetails: raceDetails,
+      racesDisabled: racesDisabled,
+    });
+  }
+
+  onRaceDetailChange(changedRace, value) {
+    let raceDetails = this.state.raceDetails;
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js]: This just needs a very simple test.

File: app/javascript/Clients/clientIdPage/RaceEthnicity/Race.test.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  beforeEach(() => {
+    component = shallow(<Race />);
+  });
+
+  it('renders RaceForm Component', () => {
+    expect(component.find('RaceForm').length).toBe(1);
+  });
+
+  it('#onRaceChange should setState the value of races, raceDetails & racesDisabled when changedRace is Unknown', () => {
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/RaceEthnicity/Race.test.js]: I'd just break the descriptions with a describe or two:

```javascript
describe('#onRaceChange', () =>
  describe('when changedRace is Unknown', () => {
    it('sets the state of races, raceDetails and racesDisabled', () => {
    });
  });
});
```

File: app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js (hunk 3 above)
Comment (RamuKammagani-osi) [reply app/javascript/Clients/clientIdPage/RaceEthnicity/Race.js]: Had a discussion with @edgibbs , We don't really need to test this, So @Sushmareddy-OSI don't need to worry about this one!

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Having some issues with really slow Jenkins builds, but it's all green.

**Pull request merged**